diff -urN aaa/metakit-2.4.9.2/unix/Makefile.in metakit-2.4.9.2/unix/Makefile.in
--- aaa/metakit-2.4.9.2/unix/Makefile.in	2003-02-08 01:52:17.000000000 +0100
+++ metakit-2.4.9.2/unix/Makefile.in	2003-08-02 17:11:37.000000000 +0200
@@ -170,11 +170,11 @@
 		-L$(libdir) -llua -llualib
 
 demo: $(srcdir)/../demos/demo.cpp libmk4.la
-	$L --mode=link $(CXX) $(CXX_SWITCHES) -static -o $@@EXEEXT@ \
+	$L --mode=link $(CXX) $(CXX_SWITCHES) -o $@@EXEEXT@ \
 		$(srcdir)/../demos/demo.cpp libmk4.la
 
 dump: $(srcdir)/../demos/dump.cpp libmk4.la
-	$L --mode=link $(CXX) $(CXX_SWITCHES) -static -o $@@EXEEXT@ \
+	$L --mode=link $(CXX) $(CXX_SWITCHES) -o $@@EXEEXT@ \
 		$(srcdir)/../demos/dump.cpp libmk4.la
 
 myio: $(srcdir)/../demos/myio.cpp libmk4.la
@@ -186,7 +186,7 @@
 		$(srcdir)/../demos/struct.cpp -lmk4
 
 regress: $(TSTOBJS) libmk4.la
-	$L --mode=link $(CXX) -static -o $@@EXEEXT@ $(TSTOBJS) libmk4.la
+	$L --mode=link $(CXX) -o $@@EXEEXT@ $(TSTOBJS) libmk4.la
 
 #---------- Dependencies
 
