diff -urN metakit-2.01.org/unix/Makefile.in metakit-2.01/unix/Makefile.in
--- metakit-2.01.org/unix/Makefile.in	Sun Apr 22 11:01:17 2001
+++ metakit-2.01/unix/Makefile.in	Sun Apr 22 11:01:51 2001
@@ -31,7 +31,7 @@
 top_builddir = .
 
 # not normally compiled with -g, but you can specify it in the make command
-CXXFLAGS = -O3 -fomit-frame-pointer -DNDEBUG
+CXXFLAGS += -fomit-frame-pointer -DNDEBUG
 #CXXFLAGS = -fguiding-decls -Wall -pedantic -Wno-unused -g
 
 CXX = @CXX@
