--- metakit-2.01.orig/include/mk4.h
+++ metakit-2.01/include/mk4.h
@@ -129,7 +129,7 @@
 
     // GNU gcc/egcs
 #if defined (__GNUC__)
-#if __bool_true_false_are_defined && !defined (q4_BOOL)
+#if !defined (q4_BOOL)
 #define q4_BOOL 1
 #endif
 #endif
--- metakit-2.01.orig/src/gnuc.h
+++ metakit-2.01/src/gnuc.h
@@ -4,6 +4,4 @@
 
 #define q4_GNUC 1
 
-#if __bool_true_false_are_defined
 #define q4_BOOL 1
-#endif
--- metakit-2.01.orig/unix/Makefile.in
+++ metakit-2.01/unix/Makefile.in
@@ -46,7 +46,7 @@
 
 CXX_SWITCHES      = $(CXXFLAGS) -I../include -I../src -I.
 CXX_SWITCHES_TCL  = $(CXXFLAGS) -I../include -I../../include \
-			-I$(TCL_BIN_DIR)/../generic -DUSE_TCL_STUBS
+			-I$(subst lib,include,$(TCL_BIN_DIR)) -DUSE_TCL_STUBS
 CXX_SWITCHES_PY   = $(CXXFLAGS) -I../include -I../python/scxx \
 			-I$(includedir)/python1.5
 

